libraries/rush-lib/src/cli/RushPnpmCommandLineParser.ts

```diff
@@ -281,6 +281,14 @@ export class RushPnpmCommandLineParser {
             );
             throw new AlreadyReportedError();
           }
+          // patch-commit internally calls installation under cwd instead of the common/temp folder
+          // It throws missing package.json error, so in this case, we need to set the dir to the common/temp folder here
+          if (pnpmArgs.indexOf('--dir') < 0 && pnpmArgs.indexOf('-C') < 0) {
+            if (!FileSystem.exists(`${process.cwd()}/package.json`)) {
```

Comment:
Unsure if this check is necessary. There are many Rush projects that have a package.json file at the root of the repo, generally to provide npm scripts at the root level. Additionally, Rush always uses the common/temp folder for the install directory for PNPM

Reply:
Probably the most expected behavior would be to always pass this when we're doing an operation that expects to be run in the simulated workspace root.

Reply:
The tricky part is the the installation just expects a "package.json" file under the working directory. After that, the lockfile directory are used to correctly install deps for all packages... where lockfile directory is set to "common/temp" before.

Reply:
If there is a `package.json` at the repo root, how does pnpm find the `common/temp` lockfile without the `--dir`/`-C` parameter?

Reply:
In pnpm, there are two "dir" settings, 1. `dir`  2. `lockfileDir`.  `lockfileDir` are used to specify where is the pnpm-lock.yaml, which is `common/temp/pnpm-lock.yaml`.

Related code:
```
    const workspaceFolder: string = rushConfiguration.commonTempFolder;
    const pnpmEnvironmentMap: EnvironmentMap = new EnvironmentMap(process.env);
    pnpmEnvironmentMap.set('NPM_CONFIG_WORKSPACE_DIR', workspaceFolder);
```